### PR TITLE
vulkan-extensionlayer 1.3.246 (new formula)

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

~~Note that the test is incomplete right now; I'm putting this PR up to get more eyes on it, and don't intend it to be merged yet.~~  Test finished!

### CODE REVIEW

File: Formula/vulkan-extensionlayer.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    ENV.prepend_path "VK_LAYER_PATH", layers_dir
+    if OS.mac?
+      ENV["VK_ICD_FILENAMES"] = Formula["molten-vk"].share/"vulkan"/"icd.d"/"MoltenVK_icd.json"
+      ENV.prepend_path "DYLD_LIBRARY_PATH", Formula["vulkan-loader"].lib
+    elsif OS.linux?
+      ENV.prepend_path "LD_LIBRARY_PATH", layers_dir
+      ENV.prepend_path "LD_LIBRARY_PATH", Formula["vulkan-loader"].lib
+    else
+      odie "Unsupported platform!"
+    end
```
Comment (chenrui333) [inline Formula/vulkan-extensionlayer.rb:70]: there is no need to have it

**Comment by pthariensflame** [comment]
@carlocab Since you've been helping with all the Vulkan stuff so far.  ❤️

File: Formula/vulkan-extensionlayer.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    depends_on "wayland" => :build
+  end
+
+  def install
+    system "cmake", "-S", ".", "-B", "build",
+           "-DGLSLANG_INSTALL_DIR=#{Formula["glslang"].prefix}",
+           "-DSPIRV_HEADERS_INSTALL_DIR=#{Formula["spirv-headers"].prefix}",
+           "-DSPIRV_TOOLS_INSTALL_DIR=#{Formula["spirv-tools"].prefix}",
+           "-DVULKAN_HEADERS_INSTALL_DIR=#{Formula["vulkan-headers"].prefix}",
+           *std_cmake_args
```
Comment (carlocab) [inline Formula/vulkan-extensionlayer.rb:33]:
```suggestion
    system "cmake", "-S", ".", "-B", "build",
                    "-DGLSLANG_INSTALL_DIR=#{Formula["glslang"].prefix}",
                    "-DSPIRV_HEADERS_INSTALL_DIR=#{Formula["spirv-headers"].prefix}",
                    "-DSPIRV_TOOLS_INSTALL_DIR=#{Formula["spirv-tools"].prefix}",
                    "-DVULKAN_HEADERS_INSTALL_DIR=#{Formula["vulkan-headers"].prefix}",
                    *std_cmake_args
```
Let's align these as arguments of `cmake` and not `system`.

File: Formula/vulkan-extensionlayer.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    expected = <<~EOS
+      Instance Layers: count = #{build.head? ? 3 : 2}
+      --------------------------
+    EOS
+    if build.head?
+      expected += <<~EOS
+        VK_LAYER_KHRONOS_shader_object      Shader object layer              #{ver}  version 1
+      EOS
+    end
+    expected += <<~EOS
```
Comment (carlocab) [inline Formula/vulkan-extensionlayer.rb:59]:
```suggestion
      Instance Layers: count = 2
      --------------------------
    EOS
```
Let's make this simpler -- we don't test `head` in CI.

Comment (pthariensflame) [reply Formula/vulkan-extensionlayer.rb:59]: Okay!  Is there a way to remind me—or whoever updates the formula next—that the next released version will introduce `VK_LAYER_KHRONOS_shader_object` and so the test will need to be updated?

Comment (pthariensflame) [reply Formula/vulkan-extensionlayer.rb:59]: Trying something for this.

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by pthariensflame** [comment]
🎉❓

**Pull request merged**